libcudacxx/include/cuda/std/detail/libcxx/include/__bit_reference

```diff
@@ -22,1272 +20,1354 @@
 #  pragma system_header
 #endif // no system header
 
-_LIBCUDACXX_PUSH_MACROS
-#include <__undef_macros>
-
+#include <cuda/std/__algorithm/copy_n.h>
+#include <cuda/std/__algorithm/fill_n.h>
+#include <cuda/std/__algorithm/min.h>
+// TODO: modularize bit a bit
+#include <cuda/std/detail/libcxx/include/bit>
+// #include <cuda/std/__bit/countr.h>
+// #include <cuda/std/__bit/invert_if.h>
+// #include <cuda/std/__bit/popcount.h>
+#include <cuda/std/__iterator/iterator_traits.h>
+#include <cuda/std/__memory/construct_at.h>
+#include <cuda/std/__memory/pointer_traits.h>
+#include <cuda/std/__type_traits/conditional.h>
+#include <cuda/std/__utility/swap.h>
+#include <cuda/std/detail/libcxx/include/cstring>
 
 _LIBCUDACXX_BEGIN_NAMESPACE_STD
 
-template <class _Cp, bool _IsConst, typename _Cp::__storage_type = 0> class __bit_iterator;
-template <class _Cp> class __bit_const_reference;
+template <class _Cp>
+class __bit_const_reference;
 
 template <class _Tp>
 struct __has_storage_type
 {
-    static const bool value = false;
+  static const bool value = false;
 };
 
+template <class _Cp, bool _IsConst>
+class __bit_iterator;
+
 template <class _Cp, bool = __has_storage_type<_Cp>::value>
 class __bit_reference
 {
-    typedef typename _Cp::__storage_type    __storage_type;
-    typedef typename _Cp::__storage_pointer __storage_pointer;
+  using __storage_type    = typename _Cp::__storage_type;
+  using __storage_pointer = typename _Cp::__storage_pointer;
 
-    __storage_pointer __seg_;
-    __storage_type    __mask_;
+  __storage_pointer __seg_;
+  __storage_type __mask_;
 
-    friend typename _Cp::__self;
+  friend typename _Cp::__self;
 
-    friend class __bit_const_reference<_Cp>;
-    friend class __bit_iterator<_Cp, false>;
-public:
-    _LIBCUDACXX_INLINE_VISIBILITY
-    __bit_reference(const __bit_reference&) = default;
-
-    _LIBCUDACXX_INLINE_VISIBILITY operator bool() const noexcept
-        {return static_cast<bool>(*__seg_ & __mask_);}
-    _LIBCUDACXX_INLINE_VISIBILITY bool operator ~() const noexcept
-        {return !static_cast<bool>(*this);}
+  friend class __bit_const_reference<_Cp>;
+  friend class __bit_iterator<_Cp, false>;
 
-    _LIBCUDACXX_INLINE_VISIBILITY
-    __bit_reference& operator=(bool __x) noexcept
+public:
+  using __container = typename _Cp::__self;
+
+  _LIBCUDACXX_HIDE_FROM_ABI _CCCL_CONSTEXPR_CXX14 __bit_reference(const __bit_reference&) = default;
```

Comment:
Question: I am of the opinion, that we should just drop all of `_LIBCUDACXX_HIDE_FROM_ABI ` as we do not use it and it turned out to be highly nontrivial to change it.

Reply:
What's the question though? :P

Reply:
Should we do that